src/Control/Biegunka/Source/Git/Internal.hs

```diff
@@ -91,41 +95,45 @@ git u p s = git' u p (actions s)
 git_ :: URI -> FilePath -> Script 'Sources ()
 git_ u p = git' u p mempty
 
-updateGit :: URI -> FilePath -> Git -> IO (Maybe String)
+updateGit :: URI -> FilePath -> Git -> IO (Maybe String, IO (Maybe String))
```

Comment:
The outer IO action looks for the changes and gives them back, if any, as string together with the inner IO action that actually applies these changes. The reason for the second `Maybe String` is that the finalizer may have more information about the changes that actually happened (for instance, when cloning repositories)